core/src/Threads/Kokkos_Threads_Parallel_Range.hpp

```diff
@@ -249,8 +234,7 @@ class ParallelReduce<FunctorType, Kokkos::RangePolicy<Traits...>, ReducerType,
         const pointer_type data =
             (pointer_type)ThreadsExec::root_reduce_scratch();
 
-        const unsigned n = Analysis::value_count(
-            ReducerConditional::select(m_functor, m_reducer));
+        const unsigned n = m_functor_reducer.get_reducer().value_count();
```

Comment:
```suggestion
        const unsigned n = reducer.value_count();
```

Reply:
`const ReducerType &reducer = m_functor_reducer.get_reducer();` is defined in this scope.